### PR TITLE
feat(inline-agg): add AnyValue accumulator

### PR DESCRIPTION
**Pull request by BABTUNA**
## Summary

Add `AggExpr::AnyValue` support to the inline grouped aggregation path in `src/daft-recordbatch/src/ops/inline_agg.rs`. Numeric dtypes only (Int8-64, UInt8-64, Float32/64); other dtypes continue to use the fallback path.

## Why

Part of #6585 item 7 (more accumulator types). Without inline AnyValue, any query mixing AnyValue with Sum/Min/Max/Count on numeric columns falls entirely back to the slower path. This keeps mixed queries on the inline path when the AnyValue target is numeric.

## Changes Made

- New `define_any_value_accum!` macro with 10 instantiations (`AnyValueAccumI8..I64`, `AnyValueAccumU8..U64`, `AnyValueAccumF32/F64`)
- 10 new variants on `AggAccumulator`
- New `AggExpr::AnyValue` arm in `try_create_accumulator` dispatching by dtype
- `can_inline_agg` extended to accept `AggExpr::AnyValue` for the same numeric dtype set as Sum/Min/Max
- 7 new tests

## Behavior

Per-group state is `Option<Option<T>>`:
- `None`: group has not been touched yet
- `Some(None)`: group locked with a null value (`ignore_nulls=false` only)
- `Some(Some(v))`: group locked with value `v`

Three branches in `update_batch`:
1. `source.null_count() == 0`: tight loop, `ignore_nulls` irrelevant, first row per group wins.
2. `ignore_nulls=true`: scan each group until a non-null value is seen; lock thereafter. If all values are null, the group stays `None` and the result is null.
3. `ignore_nulls=false`: first row wins per group, null or not.

Non-numeric AnyValue (Utf8, Binary, etc.) bails to fallback via `can_inline_agg`, preserving existing semantics.

## Test Plan

- `cargo test -p daft-recordbatch --lib inline_agg::tests`: 39/39 pass
- `cargo fmt -p daft-recordbatch` clean
- `cargo clippy -p daft-recordbatch --lib --tests` clean
- 7 new tests covering: ignore_nulls=true/false on int values with nulls; first-row-is-null edge case for both modes; Float64; Int8 (small int widening class); mixed AnyValue + Sum + Min + Max in one query

## Related Issues

Part of #6585 item 7.

### CODE REVIEW

File: src/daft-recordbatch/src/ops/inline_agg.rs [hunk 1]
```diff
@@ -492,6 +599,107 @@ fn try_create_accumulator(
...
+                    Ok(Some((
+                        AggAccumulator::AnyValueF64(AnyValueAccumF64::new(
+                            arr.clone(),
+                            ignore_nulls,
+                        )),
+                        name,
+                    )))
+                }
+                _ => Ok(None),
+            }
```
[the remark this answers is not in the document]
Comment (BABTUNA) [reply src/daft-recordbatch/src/ops/inline_agg.rs:818]: Keeping inline. Same as the Sum arm above which also bypasses the macro. Don't want to add a knob to a shared utility for one caller.

File: src/daft-recordbatch/src/ops/inline_agg.rs [hunk 2]
```diff
@@ -1614,4 +1829,109 @@ mod tests {
...
+            BoundAggExpr::try_new(AggExpr::AnyValue(resolved_col("val"), true), &schema).unwrap(),
+            BoundAggExpr::try_new(AggExpr::Sum(resolved_col("val")), &schema).unwrap(),
+            BoundAggExpr::try_new(AggExpr::Min(resolved_col("val")), &schema).unwrap(),
+            BoundAggExpr::try_new(AggExpr::Max(resolved_col("val")), &schema).unwrap(),
+        ];
+        let inline_result = rb.agg_groupby_inline(&bound_agg, &group_by).unwrap();
+        let fallback_result = rb.agg_groupby_fallback(&bound_agg, &group_by).unwrap();
+        assert_batches_equal(&inline_result, &fallback_result);
+    }
 }
```
[the remark this answers is not in the document]
Comment (BABTUNA) [reply src/daft-recordbatch/src/ops/inline_agg.rs:2977]: fixed

File: src/daft-recordbatch/src/ops/inline_agg.rs [hunk 3]
```diff
@@ -297,6 +297,103 @@ define_minmax_accum!(MaxAccumF64, Float64Type, f64, |a, b| if a.gt(&b) {
...
+                    }
+                }
+            }
+
+            fn finalize(self, name: &str) -> DaftResult<Series> {
+                let values: Vec<Option<$native>> = self
+                    .accumulators
+                    .into_iter()
+                    .map(|opt| opt.unwrap_or(None))
+                    .collect();
```
[the remark this answers is not in the document]
Comment (BABTUNA) [reply src/daft-recordbatch/src/ops/inline_agg.rs]: fixed

**Comment by BABTUNA** [comment]
@greptile re-review

**Review by colin-ho — APPROVED** [review]
Looks good to me. Post this, we should think about cleaning this up. 

It's going to become a many thousand line file.

Any ideas @BABTUNA?

**Comment by BABTUNA** [comment]
@colin-ho I got 2 possible fixes:

1st one is that we can start a refactor to move into a directory of files
inline_agg/
├── mod.rs
├── accumulators.rs
├── factory.rs
├── hash_paths.rs
├── symbolized.rs
└── tests.rs

2nd one is that we keep the file as is and try is do some trait polymorphism refactor though from some of the sample runs I'm doing it seems to make performance worse. 

Wanna try the 1st approach? I could look into other methods if u want

**Comment by colin-ho** [comment]
I think we should try #2, and see what the regression % is. And also since the accumulators are very similar, it might be worth making consolidating into a generic one.

#1 is easy, we can do that as a last resort. Though I'm not a fan of dedicated `test.rs` files in rust, I prefer inlined test mods.

**Comment by colin-ho** [comment]
Looks good, if you can resolve conflicts then we can merge!

**Comment by BABTUNA** [comment]
@colin-ho merge 
image
